sidechain/peer-fetch/src/untrusted_peer_fetch.rs

```diff
@@ -0,0 +1,56 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::error::{Error, Result};
+use itc_rpc_client::direct_client::{DirectApi, DirectClient as DirectWorkerApi};
+use itp_api_client_extensions::PalletTeerexApi;
+use its_primitives::types::ShardIdentifier;
+
+/// Fetches the untrusted peer servers
+/// FIXME: Should problably be combined with the peer fetch in
+/// service/src/worker.rs
+pub struct UntrustedPeerFetcher<NodeApi> {
```

Comment:
! Logic change since last review: 
Moved Peer fetcher (originally in the same file as the block fetcher) to a separate file & introduced a  trait for easy mocking. It gets the last worker of a shard and does not use `into_async_url` fn anymore.

Reply:
The untrusted peer fetcher fetches the untrusted URL of **one** of its peers whenever it's called. Nothing is stored here, it's a one time call per sidechain block fetch. That's acceptable because hopefully the sidechain block fetching will not occur that often. 

On the other hand, the peer fetch in worker.rs `UpdatePeers` fetches all peers from the node and stores it in the worker struct. Ideally, it should always check if there's a new peer, and only then do something... like opening a new ws connection to the peer ( the connection to the peers ws should be kept open for block gossiping. And most likely also be changed to the trusted URL and not untrusted. Depending on what we decide to do with the sidechain block caching)

So - the functionality is slightly different, but it does not make sense to have two peer fetcher, this will only lead to confusion. However, the merge of these two should be done in a separate PR, as it will need quite some thoughts how to structure this. And maybe also update the whole `Worker` struct concept...

Reply:
PS: We also have a `validateer-fetch` in `sidechain/validateer-fetch/`. This one fetches the validateers (peers) including a storage proof for the trusted side. So once again, slightly different, but somehow also the same. :)

Reply:
I think we can live with this for now - but it's good to keep in mind for a future refactoring 👍

Reply:
Thanks for the thorough explanation. I do agree that these should be merged, and also that this can be postponed.